components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/sender/VFSClientConnector.java

```diff
@@ -297,6 +299,63 @@ public boolean send(CarbonMessage carbonMessage, CarbonCallback carbonCallback,
+                                            org.wso2.carbon.messaging.Constants.DIRECTION_RESPONSE);
+                                    message.setProperty(
+                                            org.wso2.transport.file.connector.server.util.Constants.FILE_PATH,
+                                            filePath);
+                                    message.setProperty(
+                                            org.wso2.transport.file.connector.server.util.Constants.EOF,
+                                            true);
+                                } else {
+                                    message = null;
+                                }
+                                while ((line = bufferedReader.readLine()) != null) {
+                                    lastLine = line;
+                                    message = new BinaryCarbonMessage(ByteBuffer.
+                                            wrap(lastLine.getBytes(StandardCharsets.UTF_8)), true);
+                                    message.setProperty(org.wso2.carbon.messaging.Constants.DIRECTION,
+                                            org.wso2.carbon.messaging.Constants.DIRECTION_RESPONSE);
+                                    message.setProperty(
+                                            org.wso2.transport.file.connector.server.util.Constants.FILE_PATH,
+                                            filePath);
+                                    message.setProperty(
+                                            org.wso2.transport.file.connector.server.util.Constants.EOF,
+                                            true);
+                                }
+                                if (null != message) {
+                                    carbonMessageProcessor.receive(message, carbonCallback);
+                                }
+                            } else if (trailerSkipped && line != null) {
+                                boolean skipSendingLine = Boolean.parseBoolean(map.get(Constants.HEADER_PRESENT));
+                                while (line != null) {
+                                    message = new BinaryCarbonMessage(ByteBuffer.
+                                            wrap(line.getBytes(StandardCharsets.UTF_8)), true);
+                                    message.setProperty(org.wso2.carbon.messaging.Constants.DIRECTION,
+                                            org.wso2.carbon.messaging.Constants.DIRECTION_RESPONSE);
+                                    message.setProperty(
+                                            org.wso2.transport.file.connector.server.util.Constants.FILE_PATH,
+                                            filePath);
+                                    message.setProperty(org.wso2.transport.file.connector.server.util.
+                                            Constants.EOF, false);
+                                    line = bufferedReader.readLine();
```

Comment:
@dnwick we need to assign EOF as true for the last message.

Reply:
@ramindu90 this is fixed now